src/trigrams.py

```diff
@@ -0,0 +1,80 @@
+"""This module implements the trigrams algorithm for a given text file."""
+
+
+from __future__ import print_function
+import random
+import io
+import sys
+import re
+
+
+def open_file(path):
+    """Return a string of the contents of a given file."""
+    f = io.open(path)
+    data = f.read()
+    f.close()
+    return data
+
+
+def format_text(text):
+    """Format string returned from file."""
+    text = text.replace('--', ' ')  #Special case.
+    text = text.replace('\n', ' ')
+    text = re.sub('[^ a-zA-Z]', '', text)
+
+    return text.lower()
+
+
+def make_tri_dict(word_list):
+    """Make key value pairs of pairs of words to a list of words from a
```

Comment:
This is a really confusing doc string. Also, it's supposed to be a one-line doc string followed by more detailed documentation if there's more to say.